doc/reference/autoinstall-reference.rst

```diff
@@ -3,184 +3,1013 @@
+set to ``false``, it disables the drivers :ref:`screen and section<ai-drivers>`.
+
+id
+~~
+
+* **type:** string
+* **default:** identifier of the first available source.
+
+Identifier of the source to install (e.g., ``"ubuntu-server-minimal"``).
 
 .. _ai-network:
 
 network
-=======
+-------
 
-stub
+* **type:** netplan-format mapping, see below
+* **default:** DHCP on interfaces named ``eth*`` or ``en*``
+* **can be interactive:** yes
 
-.. _ai-oem:
+`Netplan-formatted <https://netplan.io/reference>`_ network configuration.
+This will be applied during installation as well as in the installed system.
+The default is to interpret the config for the install media, which runs
+DHCPv4 on any interface with a name matching "``eth*``" or "``en*``" but then
+disables any interface that does not receive an address.
 
-oem
-===
+For example, to run DHCPv6 on a particular NIC:
 
-stub
+.. code-block:: yaml
 
-.. _ai-packages:
+    network:
+      version: 2
+      ethernets:
+        enp0s31f6:
+          dhcp6: true
 
-packages
-========
+Note that because of a bug, the version of Subiquity released with 20.04 GA
+forces you to write this with an extra ``network:`` key like so:
+
+.. code-block:: yaml
 
-stub
+    network:
+      network:
+        version: 2
+        ethernets:
+          enp0s31f6:
+            dhcp6: true
+
+Later versions support this syntax too for compatibility but if you can
+assume a newer version you should use the former.
 
 .. _ai-proxy:
 
 proxy
-=====
+-----
 
-stub
+* **type:** URL or ``null``
+* **default:** no proxy
+* **can be interactive:** yes
 
-.. _ai-refresh-installer:
+The proxy to configure both during installation and for ``apt`` and for
+``snapd`` in the target system.
 
-refresh-installer
-=================
+.. _ai-apt:
 
-stub
+apt
+---
 
-.. _ai-reporting:
+* **type:** mapping
+* **default:** see below
+* **can be interactive:** yes
 
-reporting
-=========
+APT configuration, used both during the install and once booted into the target
+system.
 
-stub
+This section historically used the same format as curtin,
+`which is documented here <https://curtin.readthedocs.io/en/latest/topics/apt_source.html>`_.
+Nonetheless, some key differences with the format supported by curtin have been introduced:
 
-.. _ai-shutdown:
+- Subiquity supports an alternative format for the ``primary`` section,
+  allowing configuration of a list of candidate primary mirrors. During
+  installation, Subiquity will automatically test the specified mirrors and
+  select the first one that seems usable. This new behavior is only activated
+  when the ``primary`` section is wrapped in the ``mirror-selection`` section.
 
-shutdown
-========
+- The ``fallback`` key controls what Subiquity should do if no primary mirror
+  is usable.
 
-stub
+- The ``geoip`` key controls whether a geoip lookup is done to determine the
+  correct country mirror.
 
-.. _ai-snaps:
+The default is:
 
-snaps
-=====
+.. code-block:: yaml
 
-stub
+    apt:
```

Comment:
This code block has different spacing/tabbing than some of the earlier code blocks. Are they incorrect? (I'm not familiar with yaml, but the discrepancy looks odd to me).

Reply:
Looking at our examples autoinstall files, we mostly use a 2 space indent for yaml, so I propose that we standardize on that.

Reply:
It shall be done!